roles/openshift_autoheal/templates/objects.yml.j2

```diff
@@ -0,0 +1,136 @@
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+
+- apiVersion: v1
+  kind: Role
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - secrets
+    resourceNames:
+    - {{ openshift_autoheal_name }}-awx-credentials
+    - {{ openshift_autoheal_name }}-awx-tls
+    verbs:
+    - get
+
+- apiVersion: v1
+  kind: RoleBinding
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  roleRef:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  subjects:
+  - kind: ServiceAccount
+    name: {{ openshift_autoheal_name }}
+  userNames:
+  - system:serviceaccount:{{ openshift_autoheal_namespace }}:{{ openshift_autoheal_name }}
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}-awx-credentials
+  data:
+    username: {{ openshift_autoheal_awx_user | b64encode }}
+    password: {{ openshift_autoheal_awx_password | b64encode }}
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}-awx-tls
+  data:
+    ca.crt: {{ openshift_autoheal_awx_ca | b64encode }}
+
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}-config
+  data:
+    autoheal.yml: |-
+      {{ tmpconfig.stdout | indent(6) }}
+
+- apiVersion: apps/v1beta1
+  kind: Deployment
+  metadata:
+    labels:
+      app: {{ openshift_autoheal_name }}
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  spec:
+    selector:
+      matchLabels:
+        app: {{ openshift_autoheal_name }}
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          app: {{ openshift_autoheal_name }}
+      spec:
+        serviceAccountName: {{ openshift_autoheal_name }}
+        volumes:
+        - name: config
+          configMap:
+            name: {{ openshift_autoheal_name }}-config
+        containers:
+        - name: service
+          image: "{{ openshift_autoheal_image_prefix }}autoheal:{{ openshift_autoheal_image_version }}"
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+          - name: config
+            mountPath: /etc/autoheal
+          command:
+          - /usr/bin/autoheal
+          args:
+          - server
+          - --config-file=/etc/autoheal/autoheal.yml
+          - --logtostderr
+
+- apiVersion: v1
+  kind: Service
```

Comment:
Let's [front this with TLS](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_monitor_availability/files/monitor-app-create.yaml#L76) and put an auth proxy in front of the app process.